### DISCUSSION TITLE
Is it possible to use the rpc data for external apps?

### DISCUSSION THREAD

**Discussion by JoqniX** [pear-devs/pear-desktop#1798] [Q&A] [answered]
hi, i had a doubt! so i wanted to somehow clone the rpc data which the discord rich presence plugin does. to a different client or bot 
for example -  i use a local music stream bot - https://github.com/dandln/TheEpicAudioStreamer/ 
to locally stream my yt music stuff through a virtual cable on my discord server. it runs as a console thingy. which read's discord bot token etc. and listens to a audio device etc.. 

so i was hopping it would be somehow be possible to clone the same data which the youtubemusic app sends to my discord acc. to be also sent to this music bot somehow? ( i thought if its possible to read data of youtube music app. then it would be easier to get music data - such as songname, title, artist etc + also discord rpc data as well. so the bot needs to be configured to read those data and input em to discord through slash commands or so? ) 

i did make a feature request in their repo as well. but i wanted to know if it's somehow possible or not? 
if you guys have a different suggestion or method to do this. do lmk! 
thank you✨

**Comment by ArjixWasTaken** [comment, marked as answer]
You can have an [arRPC](https://github.com/OpenAsar/arrpc) server running locally!
Sadly [TheEpicAudioStreamer](https://github.com/dandln/TheEpicAudioStreamer/) needs to support arRPC, which ain't hard to do.

They just have to listen for messages from `ws://127.0.0.1:1337` and update the bot's rich presence based on them.

---

Note: Normal discord may conflict with arRPC, might need some testing.

**Reply by JoqniX** [reply]
thank you! 😊
